README.md

```diff
@@ -35,55 +39,127 @@ First you need to setup the CoreData stack with a specifc file name. Of course y
+
+### Models Configuration
+
+The models configuration is done as in any other projects.
+
+First create and configure your model entities inside the `.xcdatamodel` file.
+Then generate with Xcode the NSManagedObject subclasses as you are used to.
+
+After that, create category files (or extensions in Swift) for each model.
+The functions and logic will be implemented in those files. If it was done in the generated files, your changes would be removed everytime you generate them again.
+
+_Example_: `DBEntity.swift` and `DBEntity+DKDBManager.swift`
+
+__warning__ If your code is in Swift you can either generate the NSManagedObject subclasses in _Swift_ or _Obj-C_.
+
+- _Swift_: add `@objc(ClassName)` before the implementation:
+
+		@objc(Entity)
+		class Entity: NSManagedObject {
+			@NSManaged var createdAt: NSDate?
+			@NSManaged var updatedAt: NSDate?
+		}
+
+- _Obj-C_: import the class header in the bridge-header file.
+
+		#import "Entity.h"
+
+### Simple local database
+
+This part explains how to create and configure a single local database that do not need to be updated from an API.
+In this case there is no need of deprecated entities or automatic updates.
+
+As explained earlier, you first need the data inside a NSDictionary object.
+This data will get parsed and the library will apply a CRUD logic on it.
+
+In each extented class the following methods are **required**:
+
+[+ primaryPredicateWithDictionary:](http://cocoadocs.org/docsets/DKDBManager/0.5.1/Categories/NSManagedObject+DKDBManager.html#//api/name/primaryPredicateWithDictionary:) to create a predicate used in the CRUD process to find the right entity corresponding to the given dictionary.
+
+	+ (NSPredicate *)primaryPredicateWithDictionary:(NSDictionary *)dictionary {
+		// If returns nil then only ONE entity will ever be created and updated.
+		// If returns a `false predicate` then a new entity will always be created.
+		// Otherwise the CRUD process use the entity found by the predicate.
+		return NSPredicate(format: "FALSEPREDICATE");
+	}
+
+[- updateWithDictionary:](http://cocoadocs.org/docsets/DKDBManager/0.5.1/Categories/NSManagedObject+DKDBManager.html#//api/name/updateWithDictionary:) to update the current entity with a given dictionary.
+
+	override public func updateWithDictionary(dictionary: [NSObject : AnyObject]!) {
+        // Update attributes
+        self.id         = GET_NUMBER(dictionary, "id")
+        self.firstName  = GET_STRING(dictionary, "firstName")
+        self.lastName   = GET_STRING(dictionary, "lastName")
+    }
+
+The following **optional** ones are also recommended:
+
+[- description](https://developer.apple.com/library/prerelease/ios/documentation/Cocoa/Reference/Foundation/Classes/NSObject_Class/index.html#//apple_ref/occ/clm/NSObject/description) to improve how the receiving entity is described/logged.
+
+	override var description: String {
+        get {
+            return "\(self.id) : \(self.name)"
```

Comment:
maybe use `return "\(self.id) : \(self.lastName)"` to use attributes that are part of the entity.